src/chain.js

```diff
@@ -130,4 +137,23 @@ Chain.prototype.__verifyPreviousOperation = function () {
   }
 };
 
+Chain.prototype.__shouldMergeSeparatedBuffers = function (typedArrayType, operations) {
+  if (typedArrayType.indexOf('Shared') > -1) {
+    return this.__arrayReducerOperationWasApplied(operations);
+  }
+  return true;
+};
+
+Chain.prototype.__arrayReducerOperationWasApplied = function (operations) {
```

Comment:
instead of reducer (related to reduce) let's use: `__arrayChangingSizeOperationWasApplied`